deeppavlov/skills/aiml_skill/aiml_skill.py

```diff
@@ -0,0 +1,131 @@
+import random
+import re
+import os
+from typing import List, Tuple, Optional, Union
+from deeppavlov.core.skill.skill import Skill
+import aiml
+
+
+class AIMLSkill(Skill):
+    """Skill wraps python-aiml library into DeepPavlov interfrace.
+    AIML uses directory with AIML scripts which are loaded at initialization and used as patterns
+    for answering at each step.
+    """
+
+    def __init__(self,
+                 path_to_aiml_scripts: Union[str, None] = None,
+                 null_response: str = "I don't know",
+                 default_confidence: float = 0.66):
+        """
+        Construct skill:
+            read AIML scripts,
+            load AIML kernel
+
+        Args:
+            path_to_aiml_scripts: string path to folder with AIML scripts
+            null_response: Response string to answer if no AIML Patterns matched
+            default_confidence: The default confidence.
+        """
+
+        if not path_to_aiml_scripts:
+            import os
+            cur_dir = os.path.dirname(os.path.abspath(__file__))
+            path_to_aiml_scripts = cur_dir + "/aiml_scripts"
+
+        self.path_to_aiml_scripts = path_to_aiml_scripts
+
+        self.default_confidence = default_confidence
+        self.null_response = null_response
+        self.kernel = aiml.Kernel()
+        # to block AIML output:
+        self.kernel._verboseMode = False
+        self._load_scripts()
+
+    def _load_scripts(self):
+        """
+        Scripts are loaded recursively from files with extensions .xml and .aiml
+        Returns:
+
+        """
+        # learn kernel to all aimls in directory tree:
+
+        for root, dirs, files in os.walk(self.path_to_aiml_scripts):
+            for file in files:
+                if file.endswith(".xml") or file.endswith(".aiml"):
+                    file_path = os.path.join(root, file)
+                    self.kernel.learn(file_path)
+
+    def process_step(self, utterance_str: str, user_id: any):
+        response = self.kernel.respond(utterance_str, sessionID=user_id)
+        # here put your estimation of confidence:
+        confidence = self.default_confidence
+        if response:
+            # print(f"AIML responds: {response}")
+            pass
+        else:
+            # print("AIML responses silently...")
+            response = self.null_response
+        return response, confidence
+
+    def _generate_user_id(self):
+        """
+        Here you put user id generative logic if you want to implement it in the skill.
+
+        Although it is better to delegate user_id generation to Agent Layer
+        Returns: int
+
+        """
+        import datetime
+        return datetime.datetime.utcnow().timestamp() * 10e6
```

Comment:
No need for local import. You can also use [uuid module](https://docs.python.org/3.6/library/uuid.html) for id generation.

Reply:
Fixed